blockonomics.php

```diff
@@ -288,18 +334,80 @@ public function doCurlCall($url, $post_content = '')
     }
 
     public function testSetup()
+    {
+        $test_results = array();
+        $active_cryptos = $this->getActiveCurrencies();
+        foreach (array_keys($active_cryptos) as $code) {
+            $test_results[$code] = $this->testOneCrypto($code);
+        }
+        return $test_results;
+    }
+
+    /*
+     * Get list of active crypto currencies
+     */
+    public function getActiveCurrencies()
+    {
+        $active_currencies = array();
+        $blockonomics_currencies = $this->getSupportedCurrencies();
+        foreach ($blockonomics_currencies as $code => $currency) {
+            $enabled = Configuration::get('BLOCKONOMICS_' . Tools::strtoupper($code));
+            if ($enabled) {
+                $active_currencies[$code] = $currency;
+            }
+        }
+        return $active_currencies;
+    }
+
+    /*
+     * Get list of crypto currencies supported by Blockonomics
+     */
+    public function getSupportedCurrencies()
+    {
+        return array(
+              'btc' => array(
+                    'code' => 'btc',
+                    'name' => 'Bitcoin',
+                    'uri' => 'bitcoin'
+              ),
+              'bch' => array(
+                    'code' => 'bch',
+                    'name' => 'Bitcoin Cash',
+                    'uri' => 'bitcoincash'
+              )
+          );
+    }
+
+    public function testOneCrypto($crypto)
     {
         $error_str = '';
-        $url = Configuration::get('BLOCKONOMICS_GET_CALLBACKS_URL');
-        $response = $this->doCurlCall($url);
+        $response = $this->getCallbacks($crypto);
+        $error_str = $this->checkCallbackUrlsOrSetOne($crypto, $response);
+        if (!$error_str) {
+            //Everything OK ! Test address generation
+            $response = $this->getNewAddress($crypto);
```

Comment:
test_mode should be used to prevent Test Setup generating a new address with every click.
`$response = $this->getNewAddress($crypto, true);`

Reply:
Done, I've set it to true.